Sources/NIOCore/NIOTimer.swift

```diff
@@ -0,0 +1,94 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2024 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A type that handles timer callbacks scheduled with ``EventLoop/setTimer(for:_:)-5e37g``.
+///
+/// - Seealso: ``EventLoop/setTimer(for:_:)-5e37g``.
+public protocol NIOTimerHandler {
+    func timerFired(loop: any EventLoop)
+}
+
+/// An opaque handle that can be used to cancel a timer.
+///
+/// Users cannot create an instance of this type; it is returned by ``EventLoop/setTimer(for:_:)-5e37g``.
+///
+/// - Seealso: ``EventLoop/setTimer(for:_:)-5e37g``.
+public struct NIOTimer {
```

Comment:
We might want to give this a name that implies it's more of a handle.

Reply:
How strongly do you feel about this? I did try this, but personally, I didn't like the term handle for a few reasons.

Mainly because "handle" is such a generic term in an API. If you have a value of a type, isn't that always a "handle" of sorts? Along these lines, this has one method called `cancel()` and a method is a verb and the subject of the verb is the thing you have. I want to "cancel the timer" not "cancel the timer handle". If we want to use the term handle, then the method should probably called `cancelTimer()`.

Secondarily, `NIOTimerHandler` and `NIOTimerHandle` were just very similar and reading the API surface took a lot more care.

Reply:
Not very strongly.